crates/aptos-ledger/src/lib.rs

```diff
@@ -60,6 +66,128 @@ impl From<LedgerHIDError> for AptosLedgerError {
     }
 }
 
+#[derive(Debug)]
+/// Status code returned when communicating with ledger
+/// Most Aptos ones defined here https://github.com/aptos-labs/ledger-app-aptos/blob/main/doc/COMMANDS.md#status-words
+/// Some of the ledger status code defined here - https://www.eftlab.com/knowledge-base/complete-list-of-apdu-responses
+pub enum AptosLedgerStatusCode {
+    /// Aptos ledger app related status code
+
+    /// Rejected by user
+    Deny = 0x6985,
+
+    /// Either P1 or P2 is incorrect
+    WrongPip2 = 0x6A86,
+
+    /// Lc or minimum APDU length is incorrect
+    WrongDataLength = 0x6A87,
+
+    /// No command exists with INS
+    InsNotSupported = 0x6D00,
+
+    /// Bad CLA used for this application
+    ClaNotSupported = 0x6E00,
+
+    /// Wrong response length (buffer size problem)
+    WrongResponseLength = 0xB000,
+
+    /// BIP32 path conversion to string failed
+    DisplayBip32PathFail = 0xB001,
+
+    /// Address conversion to string failed
+    DisplayAddressFail = 0xB002,
+
+    /// Amount conversion to string failed
+    DisplayAmountFail = 0xB003,
+
+    /// Wrong raw transaction length
+    WrongTxnLength = 0xB004,
+
+    /// Failed to parse raw transaction
+    TxnParsingFail = 0xB005,
+
+    /// Failed to compute hash digest of raw transaction
+    TxnHashFail = 0xB006,
+
+    /// Security issue with bad state
+    BadState = 0xB007,
+
+    /// Signature of raw transaction failed
+    SignatureFail = 0xB008,
+
+    /// Success
+    Success = 0x9000,
+
+    /// Ledger device related general status code
+
+    /// There are more status code, but we only list the most common ones
+    /// Ledger device is locked
+    LedgerLocked = 0x5515,
+
+    /// Aptos ledger app is not opened
+    AppNotOpen = 0x6E01,
+
+    /// Self Defined status code for Unknown code
+    /// Unknown status code
+    Unknown = 0x0000,
+}
+
+impl AptosLedgerStatusCode {
+    fn description(&self) -> &str {
+        match self {
+            AptosLedgerStatusCode::Deny => "Request rejected by user",
+            AptosLedgerStatusCode::WrongPip2 => "Wrong P1 or P2",
+            AptosLedgerStatusCode::WrongDataLength => "Wrong data length",
```

Comment:
Instead of using the num_enum crate, I just found out that I can convert the enum to hex using follow
write!(f, "{} (0x{:x})", self.description(), *self as u16)

Which will end up with result looks like this ->
Request rejected by user (0x6985)"